src/unstract/sdk/llm.py

```diff
@@ -54,12 +55,17 @@ def _initialise(self):
         if self._adapter_instance_id:
             self._llm_instance = self._get_llm(self._adapter_instance_id)
             self._usage_kwargs["adapter_instance_id"] = self._adapter_instance_id
-            platform_api_key = self._tool.get_env_or_die(ToolEnv.PLATFORM_API_KEY)
-            CallbackManager.set_callback(
-                platform_api_key=platform_api_key,
-                model=self._llm_instance,
-                kwargs=self._usage_kwargs,
+
+            is_public_adapter = SdkHelper.is_public_adapter(
+                adapter_id=self._adapter_instance_id
             )
+            if not is_public_adapter:
```

Comment:
```suggestion
            if not SdkHelper.is_public_adapter(adapter_id=self._adapter_instance_id):
```
Please take care of line length issues if you accept this. You could extend the same to other similar usages as well